File: Php/tests/AppTests/Admin/AdminNavModelTests.php
namespace tests\AppTests\Admin{

	use tests\UnitTestCase;
	use Domain\Classes\User;
	use entities\Role;
	use app\Admin\AdminNavModel;
	use tests\TestHelper;

	class AdminNavModelTests extends UnitTestCase {

		function test_Show_AdminUser_DefaultContentSet(){
			$user = new User();
			$user.setGlobalRole(Role::$admin);
			$sut = new AdminNavModel($user);

			assertIdentical("Admin", $sut.heading);
			assertIdentical("admin-nav", $sut.cssClass);
		}

		function test_Show_NotLoggedIn_NoNodes(){
			$sut = new AdminNavModel();

			$nodes = $sut.nodes;
			assertIdentical(0, count($nodes));
		}

		function test_Show_WithNonAdminUser_NoNodes(){
			$user = new User();
			$sut = new AdminNavModel($user);

			$nodes = $sut.nodes;
			assertIdentical(0, count($nodes));
		}

		function test_Show_WithAdminUser_SetsNodes(){
			$user = new User();
			$user.setGlobalRole(Role::$admin);
			$sut = new AdminNavModel($user);

			$nodes = $sut.nodes;
			assertIsA($nodes[0].urlModel, 'app\Urls\HomegameListingUrlModel');
			assertIsA($nodes[1].urlModel, 'app\Urls\UserListingUrlModel');
		}

	}

}

File: Web.Tests/ModelTests/NavigationModels/AdminNavModelTests.cs
using Core.Classes;
using NUnit.Framework;
using Web.Models.NavigationModels;
using Web.Models.UrlModels;

namespace Web.Tests.ModelTests.NavigationModels{

	public class AdminNavModelTests
	{
	    private User _user;

        [SetUp]
        public void SetUp()
        {
            _user = null;
        }

        [Test]
		public void Show_AdminUser_DefaultContentSet(){
			_user = new User {GlobalRole = Role.Admin};
            var sut = GetSut();

			Assert.AreEqual("Admin", sut.Heading);
			Assert.AreEqual("admin-nav", sut.CssClass);
		}

		[Test]
		public void Show_NotLoggedIn_NoNodes(){
			var sut = GetSut();

			Assert.AreEqual(0, sut.Nodes.Count);
		}

		[Test]
		public void Show_WithNonAdminUser_NoNodes(){
			_user = new User();
			var sut = GetSut();

			Assert.AreEqual(0, sut.Nodes.Count);
		}

		[Test]
		public void Show_WithAdminUser_SetsNodes(){
			_user = new User {GlobalRole = Role.Admin};
		    var sut = GetSut();

			Assert.IsInstanceOf<HomegameListingUrlModel>(sut.Nodes[0].UrlModel);
			Assert.IsInstanceOf<UserListingUrlModel>(sut.Nodes[1].UrlModel);
		}

        private AdminNavModel GetSut()
        {
            return new AdminNavModel(_user);
        }

	}

}